### PR TITLE
Avoid Native Queries

### LINKED ISSUE #222: Revert native sql unlock before we ship the refactored access management
### Please agree to the following

- [X] I have searched [existing issues](https://github.com/cryptomator/hub/issues?q=) for duplicates
- [X] I agree to follow this project's [Code of Conduct](https://github.com/cryptomator/hub/blob/develop/.github/CODE_OF_CONDUCT.md)

### Summary

Revert https://github.com/cryptomator/hub/commit/1baf7340870093676e7476331b6e94e417917518 before we ship the refactored access management to remove the native call which can introduce major problems in the future.

### Motivation

Make sure that the introduced native unlock call https://github.com/cryptomator/hub/commit/1baf7340870093676e7476331b6e94e417917518 never makes it into production. I strongly suspect that Hibernate ORM will be at version 6.2.8 or 6.3.0 until we ship the refactored access management.

This issue should just remind us to undo that commit before shipping.

### Considered Alternatives

We should not downgrade Quarkus to v3.1.0 because of https://github.com/cryptomator/hub/issues/220

Downgrading only the hibernate dependency wasn't successfully. 

### Anything else?

Regarding the issue itself, see https://github.com/quarkusio/quarkus/issues/35386

### PR DESCRIPTION
**Pull request by overheadhunter**
Fixes #222 by reverting native queries back to `@NamedQuery`.

### AccessToken

The unlock `@NamedQuery` avoids any join on User:

```sql
SELECT token
FROM AccessToken token
INNER JOIN EffectiveVaultAccess perm ON token.id.vaultId = perm.id.vaultId AND token.id.userId = perm.id.authorityId
WHERE token.id.vaultId = :vaultId AND token.id.userId = :userId
```

Hibernate translates this to the following:

```sql
    select
        a1_0."user_id",
        a1_0."vault_id",
        a1_0."vault_masterkey" 
    from
        "access_token" a1_0 
    join
        "effective_vault_access" e1_0 
            on a1_0."vault_id"=e1_0."vault_id" 
            and a1_0."user_id"=e1_0."authority_id" 
    where
        a1_0."vault_id"=? 
        and a1_0."user_id"=? fetch first ? rows only
```

### LegacyAccessToken

This is the `@NamedQuery`:

```sql
SELECT token
FROM LegacyAccessToken token
INNER JOIN LegacyDevice device ON device.id = token.id.deviceId
INNER JOIN EffectiveVaultAccess perm ON token.id.vaultId = perm.id.vaultId AND device.ownerId = perm.id.authorityId
WHERE token.id.vaultId = :vaultId AND token.id.deviceId = :deviceId AND device.ownerId = :userId
```

Hibernate translates this to the following:

```sql
    select
        l1_0."device_id",
        l1_0."vault_id",
        l1_0."jwe" 
    from
        "access_token_legacy" l1_0 
    join
        "device_legacy" l2_0 
            on l2_0."id"=l1_0."device_id" 
    join
        "effective_vault_access" e1_0 
            on l1_0."vault_id"=e1_0."vault_id" 
            and l2_0."owner_id"=e1_0."authority_id" 
    where
        l1_0."vault_id"=? 
        and l1_0."device_id"=? 
        and l2_0."owner_id"=?
```

## Summary by CodeRabbit

- **Refactor**
	- Updated access token retrieval to use a named query for improved maintainability.
	- Transitioned from native SQL to JPQL in `LegacyAccessToken` to enhance database compatibility.
- **New Features**
	- Introduced an `ownerId` field to `LegacyDevice` for better ownership tracking.

### CODE REVIEW

**Comment by overheadhunter** [comment]
@coderabbitai review

**Comment by overheadhunter** [comment]
@coderabbitai review

File: backend/src/main/java/org/cryptomator/hub/entities/LegacyDevice.java [hunk 1]
```diff
@@ -15,4 +15,7 @@ public class LegacyDevice extends PanacheEntityBase {
 	@Column(name = "id", nullable = false)
 	public String id;
 
+	@Column(name = "owner_id", nullable = false)
+	public String ownerId;
```
[the remark this answers is not in the document]
Comment (overheadhunter) [reply backend/src/main/java/org/cryptomator/hub/entities/LegacyDevice.java:19]: added comment in b1161ae6a94e3b1bc8cf7218b303322c728cf937

**Review by SailReal — APPROVED** [review]
LGTM, nice improvement and thanks for fixing :)

File: backend/src/main/java/org/cryptomator/hub/entities/LegacyDevice.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (overheadhunter) [reply backend/src/main/java/org/cryptomator/hub/entities/LegacyDevice.java:19]: The comment has not been added to the specific field but rather to the class body, as I wanted to explain the purpose of all the included as well as excluded fields, not just the purpose of `ownerId`.

**Pull request merged**